src/modules/thirdParty/sagas.spec.ts

```diff
@@ -256,3 +277,256 @@ describe('when fetching third party available slots', () => {
     })
   })
 })
+
+describe('when publishing third party items', () => {
+  let collection: Collection
+  let item: Item
+  beforeEach(() => {
+    collection = { name: 'valid collection name' } as Collection
+    item = { ...mockedItem, collectionId: 'a valid collectionId' }
+  })
+
+  describe('when the api request fails', () => {
+    let errorMessage: string
+
+    beforeEach(() => {
+      errorMessage = 'Some Error Message'
+    })
+
+    it('should put the publish third party items fail action with an error', () => {
+      return expectSaga(thirdPartySaga, mockBuilder)
+        .provide([
+          [select(getCollection, item.collectionId), collection],
+          [call(getPublishItemsSignature, thirdParty.id, 1), { signature: '', signedMessage: '', salt: '' }],
+          [matchers.call.fn(mockBuilder.publishTPCollection), throwError(new Error(errorMessage))]
+        ])
+        .put(publishThirdPartyItemsFailure(errorMessage))
+        .dispatch(publishThirdPartyItemsRequest(thirdParty, [item]))
+        .run({ silenceTimeout: true })
+    })
+  })
+
+  describe('when the api request succeeds', () => {
+    let itemCurations: ItemCuration[]
+    beforeEach(() => {
+      collection = { name: 'valid collection name', id: uuidv4() } as Collection
+      itemCurations = [
+        {
+          id: 'id',
+          itemId: 'itemId',
+          createdAt: 0,
+          status: CurationStatus.PENDING,
+          updatedAt: 0
+        }
+      ]
+    })
+
+    it('should put the fetch third party success action with the new itemCurations and close the PublishThirdPartyCollectionModal modal', () => {
+      const mockedItemReturnedByServer = { ...mockedItem, id: 'a new id' }
+      return expectSaga(thirdPartySaga, mockBuilder)
+        .provide([
+          [select(getCollection, item.collectionId), collection],
+          [call(getPublishItemsSignature, thirdParty.id, 1), { signature: '', signedMessage: '', salt: '' }],
+          [matchers.call.fn(mockBuilder.publishTPCollection), { items: [mockedItemReturnedByServer], itemCurations }]
```

Comment:
The matchers of the API calls are not using `call` so their parameters aren't checked, would you mind changing them? At least for the success cases.

Reply:
Ohhh, sorry @LautaroPetaccio that's what you meant in your last message. OK, got it, everything updated to use the actual `call` and not the matchers.